src/api/CompileTools.ts

```diff
@@ -435,7 +435,15 @@ export namespace CompileTools {
 
                 // Then we download the files that is specified.
                 const downloads = chosenAction.postDownload.map(
-                  downloadPath => clinet.getFile(path.join(localDir, downloadPath), path.posix.join(remoteDir, downloadPath))
+                  downloadPath => {
+                    const localPath = path.join(localDir, downloadPath);
+                    const remotePath = path.posix.join(remoteDir, downloadPath);
+                    if (downloadPath.endsWith(`/`)) {
+                      return client.getDirectory(localPath, remotePath)
+                    } else {
+                      return client.getFile(localPath, remotePath)
+                    }
```

Comment:
I'd suggest testing if the remote path is actually a directory, so we don't just trust how the path ends, and uses don't have to worry about it.
```suggestion
                    const isDirectory = (await connection.sendCommand({
                        command: `[ -d "${remotePath}" ]`
                    }))?.code === 0;
                    if (isDirectory) {
                      return client.getDirectory(localPath, remotePath)
                    } else {
                      return client.getFile(localPath, remotePath)
                    }
```

Reply:
Can't await in the `map` here. 😢 

I like your idea, perhaps we can determine this upfront before the map.

Reply:
How about this:
```typescript
// Then we download the files that is specified.
const downloads = chosenAction.postDownload.map(
  async (downloadPath) => {
    const localPath = path.join(localDir, downloadPath);
    const remotePath = path.posix.join(remoteDir, downloadPath);
    const isDirectory = (await connection.sendCommand({
      command: `[ -d "${remotePath}" ]`
    }))?.code === 0;
    if (isDirectory) {
      return client.getDirectory(localPath, remotePath)
    } else {
      return client.getFile(localPath, remotePath)
    }
  }
);
```

Reply:
@sebjulliand 

I tried this and this snippet is always returning `code` as `1`, so it doesn't work. I am afraid I don't know this bash syntax well enough to understand it.

```
                    const isDirectoryCall = (await connection.sendCommand({
                      command: `[ -d "${remotePath}" ]`
                    }));
```

Reply:
`[ -d "${remotePath}" ]` tests if the path is a directory and returns 0 if it is.
If you open a shell and try `[ -d "/tmp" ] && echo it's a directory` it should echo `it's a directory`.

I tried it before submitting the suggestion, using `connection.sendCommand`, and got the expected result (i.e. 0 for a directory and 1 for a file).

How did it look in the output for you?!

Reply:
You can see here what the value of `remotePath` is followed by the output of `sendCommand` with that syntax: 

<img width="533" alt="image" src="https://user-images.githubusercontent.com/3708366/227735996-0a422cb1-70ca-47ef-b071-941bbfe0639c.png">

Reply:
![image](https://user-images.githubusercontent.com/11096890/227736070-d260c0d6-074a-4828-ad1a-680942387241.png)

Reply:
<img width="540" alt="image" src="https://user-images.githubusercontent.com/3708366/227736146-f92dfbb0-bc5b-4f80-b94d-e335b5f403b0.png">

Reply:
Oh I got it! The path is not resolved when used in a command. Which means that this:
`/xxx/yyy/zzz/.logs` does not get resolved to `/xxx/yyy/zzz/logs`. The command will try to check if the file/directory actually named `.logs` in `/xxx/yyy/zzz` is a directory.

Here's an alternative: run `cd ${remotePath}` and check if the return code is 0. This should work, even for symbolic links actually!.

Reply:
(sorry, I forgot the quotes up there: `[ -d "/tmp" ] && echo "it's a directory"`)

Reply:
The `cd` worked :) New commit incoming